x-pack/plugins/security_solution/public/common/lib/telemetry/events/entity_analytics/types.ts

```diff
@@ -53,76 +68,102 @@ export interface ReportAssetCriticalityCsvPreviewGeneratedParams {
   };
 }
 
-export interface ReportAssetCriticalityCsvImportedParams {
+interface ReportAssetCriticalityCsvImportedParams {
   file: {
     size: number;
   };
 }
 
-export interface ReportEntityStoreEnablementParams {
+interface ReportEntityStoreEnablementParams {
   timestamp: string;
   action: 'start' | 'stop';
 }
 
-export interface ReportEntityStoreInitParams {
+interface ReportEntityStoreInitParams {
   timestamp: string;
 }
 
-export type ReportEntityAnalyticsTelemetryEventParams =
-  | ReportEntityDetailsClickedParams
-  | ReportEntityAlertsClickedParams
-  | ReportEntityRiskFilteredParams
-  | ReportToggleRiskSummaryClickedParams
-  | ReportRiskInputsExpandedFlyoutOpenedParams
-  | ReportAddRiskInputToTimelineClickedParams
-  | ReportAssetCriticalityCsvPreviewGeneratedParams
-  | ReportAssetCriticalityFileSelectedParams
-  | ReportAssetCriticalityCsvImportedParams
-  | ReportEntityStoreEnablementParams
-  | ReportEntityStoreInitParams;
+// Mapping for Entity events
+export type EntityEventTypeData = {
+  [K in EntityEventTypes]: K extends EntityEventTypes.EntityDetailsClicked
```

Comment:
Not a blocker, but I'm sure there's an easier way to do this.

If we just have a record type (with the `const` modifier) where the keys are the event types (the enum) then the values can be the params types. Something like:

```
type Events = {
  EntityDetailsClicked: ReportEntityDetailsClickedParams
  ...
}

type EventType = keyof Events
```

And I think we can still use the current enum as the key, in order to preserve the nice string value as the message.